test/tests/functions/parsed/tests

```diff
@@ -1,5 +1,12 @@
 [Tests]
   design = 'source/functions/MooseParsedFunction.md'
+  [./nl_vars]
+    type = RunException
+    input = 'nl_vars_error.i'
+    expect_err = "The variables in ParsedFunction must be linear, and var 'u' is non-linear."
+    requirement = "The ParsedFunction object should error if suipplied a non-linear variable in the 'vars' field."
```

Comment:
update both of these. 

non-linear is nonlinear in moose
but here we mostly want to say field variable (as opposed to scalar variable)

also we tried to avoid saying the names of parameters in requirements. They should be readable by someone who doesnt work with MOOSE, a SQA auditor.

It could be something like:
The system should error if a field variable is passed in a parsed expression to the parsed function object, as this is currently not supported"

Reply:
Gotcha, thanks this is the kind of stuff I wouldn't have known about